src/App.tsx

```diff
@@ -1,26 +1,216 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import Header from './components/Header';
+import TodoList from './components/TodoList';
+import Footer from './components/Footer';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState('all');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  // Реф для фокусування на полі вводу
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      setLoading(true);
+      setError(null);
+
+      try {
+        const todosFromApi = await getTodos();
+
+        setTodos(todosFromApi);
+      } catch (apiError) {
+        setError('Unable to load todos');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (USER_ID) {
+      loadTodos();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      const timer = setTimeout(() => {
+        setError(null);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [error]);
+
+  // Фільтрація todos
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'active') {
+      return !todo.completed;
+    }
+
+    if (filter === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    // Перевіряємо, чи заголовок не порожній
+    if (!newTodoTitle.trim()) {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    const trimmedTitle = newTodoTitle.trim();
+
+    setIsSubmitting(true);
+
+    // Створюємо тимчасову задачу `tempTodo`
+    const newTempTodo: Todo = {
+      id: Date.now(),
+      userId: USER_ID,
+      title: trimmedTitle,
+      completed: false,
+    };
+
+    // Додаємо `tempTodo` до списку
+    setTodos(prevTodos => [...prevTodos, newTempTodo]);
+    setTempTodo(newTempTodo); // Встановлюємо тимчасову задачу
```

Comment:
do not add newTempTodo to array of todos.
It should render separately.
It is a reason of such view
![Screenshot 2024-10-29 at 14 25 19](https://github.com/user-attachments/assets/e500a69b-9b99-49ff-af93-2190ece94135)